Source/Core/Core/IOS/ES/ES.cpp

```diff
@@ -322,65 +328,89 @@ void ES::DoState(PointerWrap& p)
   }
 }
 
+std::array<ES::Context, 3>::iterator ES::FindActiveContext(u32 fd)
+{
+  return std::find_if(m_contexts.begin(), m_contexts.end(),
+                      [fd](const auto& context) { return context.ipc_fd == fd && context.active; });
+}
+
 ReturnCode ES::Open(const OpenRequest& request)
 {
-  m_caller_uid = request.uid;
-  m_caller_gid = request.gid;
+  auto context =
+      std::find_if(m_contexts.begin(), m_contexts.end(), [](const auto& c) { return !c.active; });
```

Comment:
I meant to point this out with the other find stuff, but I guess I forgot to actually comment on it (I should probably not review code at lunch time at work), but maybe extract this to a `FindInactiveContext`, for symmetry with `FindActiveContext`.